python/plugins/processing/algs/qgis/ExecuteSQL.py

```diff
@@ -125,7 +129,20 @@ def processAlgorithm(self, parameters, context, feedback):
 
         df = QgsVirtualLayerDefinition()
         for layerIdx, layer in enumerate(layers):
-            df.addSource('input{}'.format(layerIdx + 1), layer.id())
+
+            # Issue https://github.com/qgis/QGIS/issues/24041
+            # When using this algorithm from the graphic modeler, it may try to
+            # access (thanks the QgsVirtualLayerProvider) to memory layer that
+            # belongs to temporary QgsMapLayerStore, not project.
+            # So, we write them to disk is this is the case.
+            if not QgsProject.instance().mapLayer(layer.id()):
+                tf = tempfile.NamedTemporaryFile(suffix=".gpkg")
```

Comment:
You should use [`QgsVectorFileWriter::ogrDriverList`](https://qgis.org/api/classQgsVectorFileWriter.html#ab959cfe625a8dc4a3528d0451af5929d) since OGR may not be compiled with GPKG, see https://github.com/qgis/QGIS/pull/9037

Reply:
Done a25680f127